octopusdeploy/schema_runbook.go

```diff
@@ -0,0 +1,182 @@
+
+	if v, ok := d.GetOk("multi_tenancy_mode"); ok {
+		runbook.MultiTenancyMode = core.TenantedDeploymentMode(v.(string))
+	}
+
+	if v, ok := d.GetOk("connectivity_policy"); ok {
+		runbook.ConnectivityPolicy = expandConnectivityPolicy(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("environment_scope"); ok {
+		runbook.EnvironmentScope = v.(string)
+	}
+
+	if v, ok := d.GetOk("environments"); ok {
+		runbook.Environments = getSliceFromTerraformTypeList(v)
+	}
+
+	if v, ok := d.GetOk("default_guided_failure_mode"); ok {
+		runbook.DefaultGuidedFailureMode = v.(string)
+	}
+
+	if v, ok := d.GetOk("retention_policy"); ok {
+		runbook.RunRetentionPolicy = expandRunbookRetentionPolicy(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("force_package_download"); ok {
+		runbook.ForcePackageDownload = v.(bool)
+	}
+
+	return runbook
+}
+
+func getRunbookSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"id": getIDSchema(),
+		"name": {
+			Description:      "The name of the project in Octopus Deploy. This name must be unique.",
+			Required:         true,
+			Type:             schema.TypeString,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+		},
+		"description": {
+			Computed:    true,
+			Description: "The description of this runbook.",
+			Optional:    true,
+			Type:        schema.TypeString,
+		},
+		"project_id": {
+			Description: "The project that this runbook belongs to.",
+			Required:    true,
+			Type:        schema.TypeString,
+		},
+		"runbook_process_id": {
+			Description: "The runbook process ID.",
+			Computed:    true,
+			Type:        schema.TypeString,
+		},
+		"published_runbook_snapshot_id": {
+			Description: "The published snapshot ID.",
+			Computed:    true,
+			Type:        schema.TypeString,
+		},
+		"space_id": {
+			Computed:         true,
+			Description:      "The space ID associated with this project.",
+			Optional:         true,
+			Type:             schema.TypeString,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+		},
+		"multi_tenancy_mode": getTenantedDeploymentSchema(),
+		"connectivity_policy": {
+			Computed: true,
+			Elem:     &schema.Resource{Schema: getConnectivityPolicySchema()},
+			MaxItems: 1,
+			Optional: true,
+			Type:     schema.TypeList,
+		},
+		"environment_scope": {
+			Description: "Determines how the runbook is scoped to environments.",
+			Computed:    true,
+			Optional:    true,
+			Type:        schema.TypeString,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{
+				"All",
+				"Specified",
+				"FromProjectLifecycles",
+			}, false)),
+		},
+		"environments": {
+			Description: "When environment_scope is set to \"Specified\", this is the list of environments the runbook can be run against.",
+			Computed:    true,
+			Optional:    true,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Type:        schema.TypeList,
+		},
+		"default_guided_failure_mode": {
```

Comment:
should we set a default here?

Reply:
`Computed` values can not be used with a `Default` value (see [here](https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors#computed)).

I guess the real question is whether this should be a computed value.  `schema_project.go` lists the same property as `Computed`, which has been used here for consistency.

Reply:
Yeah, agree, it feels like it should not be computed, and instead we set a default.